plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java

```diff
@@ -0,0 +1,317 @@
+package com.google.cloud.tools.eclipse.appengine.newproject.maven;
+
+import com.google.cloud.tools.eclipse.appengine.newproject.AppEngineProjectIdValidator;
+import com.google.cloud.tools.eclipse.appengine.newproject.JavaPackageValidator;
+
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.jface.resource.ImageDescriptor;
+import org.eclipse.jface.wizard.IWizardPage;
+import org.eclipse.jface.wizard.WizardPage;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.ModifyEvent;
+import org.eclipse.swt.events.ModifyListener;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.DirectoryDialog;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.ui.plugin.AbstractUIPlugin;
+
+import java.text.MessageFormat;
+
+/**
+ * UI to collect all information necessary to create a new Maven-based App Engine Standard Java
+ * project.
+ */
+public class MavenAppEngineStandardWizardPage extends WizardPage implements IWizardPage {
+
+  private String defaultVersion = "0.1.0-SNAPSHOT";
+
+  private Button useDefaults;
+  private Text locationField;
+  private Button locationBrowseButton;
+  private Text groupIdField;
+  private Text artifactIdField;
+  private Text versionField;
+  private Text javaPackageField;
+  private Text projectIdField;
+  
+  public MavenAppEngineStandardWizardPage() {
+    super("basicNewProjectPage"); //$NON-NLS-1$
+    setTitle("Maven-based App Engine Standard Project");
+    setDescription("Create new Maven-based App Engine Standard Project");
+    
+    // todo get a UI designer to pick a better icon (the little plane?) and 
+    // add it to this plugin's icons folder
+    ImageDescriptor descriptor = AbstractUIPlugin.imageDescriptorFromPlugin(
+        "com.google.cloud.tools.eclipse.appengine.localserver", "icons/gcp-32x32.png"); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
Going off on a completely irrelevant tangent, it occurs to me that it would be nice if methods could declare their arguments non-NLS. Or even if there were separate types for user-visible UI strings and internal strings in the code. Nothing you can do about it here of course.